packages/utils/src/async.ts

```diff
@@ -75,3 +75,25 @@ export async function concurrentMap<A, B>(
   }
   return res
 }
+
+/**
+ * Map an async function over the values in Object x with a given concurrency level
+ *
+ * @param concurrency number of `mapFn` concurrent executions
+ * @param x associative array of values
+ * @param mapFn mapping function
+ */
+export async function concurrentValuesMap(
```

Comment:
you can probably type this as:
```
function concurrentValuesMap<IN extends any, OUT extends any>(
  concurrency:number, 
  x: Record<string, IN>
   mapFn: (val: IN, key: string) => Promise<OUT>): Promise<Record<string,OUT>
```

also it looks like `mapFn` should receive the `key` not the index in the array

Reply:
Great suggestions!